controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -755,8 +732,8 @@ func (pp *portPublisher) endpointSliceToAddresses(es *discovery.EndpointSlice) A
 					continue
 				}
 
-				for k, v := range endpoint.Topology {
-					address.TopologyLabels[k] = v
+				if endpoint.Hints != nil {
+					address.ForZones = endpoint.Hints.ForZones
```

Comment:
Very nitpicky, but I wonder if it's better to clone this here? Not sure there are any real performance implications since the address set should be shortlived (and I assume garbage cleaned at some point).